packages/authenticator/amplify_authenticator/lib/src/widgets/totp/totp_open_app_button.dart

```diff
@@ -0,0 +1,85 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'package:amplify_authenticator/amplify_authenticator.dart';
+import 'package:amplify_authenticator/src/utils/responsive.dart';
+import 'package:amplify_authenticator/src/widgets/form_field.dart';
+import 'package:amplify_flutter/amplify_flutter.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:url_launcher/url_launcher.dart';
+
+/// This widget performs an async check on the platform OS to check if the
+/// otpAuth link can be opened. Then Conditionally displays the open button
+mixin OpenAuthenticationAppButton<FieldType extends Enum,
+        T extends AuthenticatorFormField<FieldType, String>>
+    on AuthenticatorFormFieldState<FieldType, String, T> {
+  final logger = AWSLogger().createChild('Open Authenticator App Button');
+
+  /// A link which opens the registered Authenticator app.
+  /// This will open the app, but not add the account.
+  /// Used on confirm totp screen after user has already setup totp
+  static final _authenticatorAppLink = Uri.parse('otpauth://');
+
+  Uri get totpUri {
+    return switch (state.currentStep) {
+      AuthenticatorStep.continueSignInWithTotpSetup => state.totpSetupUri!,
+      _ => _authenticatorAppLink,
+    };
+  }
+
+  String get buttonText {
+    return stringResolver.buttons.resolve(
+      context,
+      ButtonResolverKey.openTotpApp,
+    );
+  }
+
+  Future<void> _launchUrl() async {
+    // If the url can be launched, then launch it and return
+    try {
+      await launchUrl(totpUri);
+    } on PlatformException catch (e, st) {
+      logger.error('Could not open Authenticator app.', e, st);
+      // If the widget is no longer mounted, then return
+      if (!context.mounted) return;
+
+      // Otherwise, show a error snack bar
+      ScaffoldMessenger.of(context).showSnackBar(
+        SnackBar(
+          backgroundColor: Colors.red[900],
+          content: const Text(
+            'Could not open Authenticator app.',
+          ),
+        ),
+      );
+    }
+  }
+
+  @override
+  Widget buildFormField(BuildContext context) {
+    final show = isMobileOrWebMobile(context);
```

Comment:
Will this show on mobile/tablet web? If yes, will it work?

Reply:
Short answer: No it won't show up because it can not open a user's app. 

Longer answer: Regardless if this statement returns true, we also check if we can launch the URI. This check is performed in the future builder below this snippet by the method `canLaunch()`. From my tests locally on simulators, the button doesn't show up in browsers on mobile devices.